src/axom/sidre/tests/spio/spio_parallel.hpp

```diff
@@ -420,6 +420,174 @@ TEST(spio_parallel, external_writeread)
   delete ds2;
 }
 
+//------------------------------------------------------------------------------
+TEST(spio_parallel, external_piecemeal_writeread)
+{
+  if(PROTOCOL != "sidre_hdf5")
```

Comment:
I have a version of this test in my work for #1479 , so that will probably replace this in the future, possibly incorporating some pieces of what you have here.